mlflow/catboost.py

```diff
@@ -86,6 +87,9 @@ def save_model(
                      or `CatBoostRegressor`_) to be saved.
     :param path: Local path where the model is to be saved.
     :param conda_env: {{ conda_env }}
+    :param code_paths: A list of local filesystem paths to Python file dependencies (or directories
+                       containing file dependencies). These files are *prepended* to the system
+                       path when the model is loaded.
```

Comment:
@stevenchen-db Can we clarify that these files are only prepended to the system path when the model is loaded as pyfunc via `mlflow.pyfunc.load_model()`?

Reply:
Are you referring to all flavors or catboost specifically? Is there any reason why the flavor specific load_model shouldn't prepend the code_paths to the system path?

Reply:
Oh, I didn't realize we were doing this for the flavor-specific case. `code` is an attribute of the `pyfunc` flavor specification of a model, so it's kind of strange that we're using / loading it when a model is restored with other flavors, e.g. `sklearn`. However, this feature seems pretty useful for users. How hard would it be to add the `code` property to each individual flavor spec?

Reply:
@dbczumar Ah I see, I was going off of the way `load_model` is handled for pytorch and noticed that the pyfunc conf gets read to set the code path. For serving, we will need `code` to be in pyfunc since we use `mlflow.pyfunc.load_model`, which already has the logic to add code paths to the system path.

Reply:
@stevenchen-db - maybe it makes sense to add code to both the pyfunc and flavor for each model and when the flavor is using flavor-specific logging, it can refer to the one in the flavor portion of MLModel?

Reply:
@ankit-db is there a scenario where the `code` attribute would be different between pyfunc and the flavor? I thought the workflow would be 1) user calls `mlflow.<flavor>.log_model` and then 2) serving will use `mlflow.pyfunc.load_model`, so whatever code was logged through the flavor should be loaded back from the pyfunc conf right?

Reply:
Oh @dbczumar were you asking if we should log `code` in both pyfunc and the flavor? I guess it might be cleaner to read `code` from the flavor conf, but I think the flavor's `code` conf would also always be a duplicate of the pyfunc `code` conf

Reply:
@stevenchen-db Yep! Ideally, it would be nice to log to both, even if the content is duplicated. This helps preserve the flavor spec abstraction.

Reply:
Makes sense, just logged both and modified `load_model`!